.github/workflows/js.yml

```diff
@@ -90,6 +90,14 @@ jobs:
         uses: actions/checkout@v3
         with:
           fetch-depth: 0
+      - name: Jest Cache
+        uses: actions/cache@v3
+        with:
+          path: js/.jest-cache
+          key: js-jest-cache-${{ runner.os }}-${{ matrix.node-version }}-${{ hashFiles('js/yarn.lock') }}
+          restore-keys: |
+            js-jest-cache-${{ runner.os }}-${{ matrix.node-version }}-
+            js-jest-cache-${{ runner.os }}-
```

Comment:
Do we need this?
(Can we reuse cache for different Node.js version?)

Reply:
The cache is reused even if different Node.js versions are used.
I have confirmed this by switching Node.js versions locally.

Reply:
Then, can we remove `-${{ matrix.node }}` from `key` and `restore-keys`?

Reply:
Will you be testing on more than one `node` in the future?
(like `node: [18, 20]`.)

I just tested a few times locally and it seemed a little faster to use the cache of the same Node.js version.
It is certainly unnecessary if you don't test with multiple `node`.

Reply:
We'll not use multiple Node.js versions unless we reduce CI time. (40+ minutes is too long to run multiple jobs. If we want to test against multiple Node.js versions, we'll use nightly CI that are defined in `dev/tasks/tasks.yml`.)

If the difference between with/without `-${{ matrix.node }}`, we want to unify it to reduce cache size. apache/arrow has multiple language implementations. So cache limit provided by GitHub Actions is small for us. So we want to reduce cache size as much as possible.

Reply:
I got it.
I removed `matrix.node`.